File: cs/ch17-03-decimal/Money/Expression.cs
using System.Net;

namespace TheSoftwareGorilla.TDD.Money;

public abstract class Expression
{
    public abstract Money Reduce(Bank bank, string to);
    public virtual Expression Plus(Expression addend) {
        return new Sum(this, addend);
    }
    public abstract Expression Times(int multiplier);
}

File: java/ch17-03-decimal/src/main/java/com/thesoftwaregorilla/tdd/money/Expression.java
package com.thesoftwaregorilla.tdd.money;

import java.math.BigDecimal;

public abstract class Expression {
    public abstract Money reduce(Bank bank, String to);
    public Expression plus(Expression addend) {
        return new Sum(this, addend);
    }
    public abstract Expression times(BigDecimal multiplier);
}

